honeybee_radiance/properties/_base.py

```diff
@@ -160,3 +163,192 @@ def ToString(self):
 
     def __repr__(self):
         return 'Base Radiance Properties:\n host: {}'.format(self.host.identifier)
+
+
+class _DynamicRadianceProperties(_RadianceProperties):
+    """Base class of Radiance Properties for all planar geometry objects.
+
+    This includes Apertures and Shades.
+
+    Args:
+        host: A honeybee_core object that hosts these properties.
+        modifier: A Honeybee Radiance Modifier for the object. If None,
+            it will be set by the parent Room ModifierSet or the Honeybee
+            default generic ModifierSet.
+        modifier_blk: A Honeybee Radiance Modifier to be used for this object
+            in direct solar simulations and in isolation studies (assessing
+            the contribution of individual Apertures).
+        dynamic_group_identifier: An optional string to note the dynamic group
+            to which the object is a part of. Objects sharing the same
+            dynamic_group_identifier will have their states change in unison.
+            If None, the object is assumed to be static.
+
+    Properties:
+        * host
+        * modifier
+        * modifier_blk
+        * dynamic_group_identifier
+        * states
+        * is_opaque
+        * is_modifier_set_on_object
+        * is_blk_overridden
+    """
+
+    __slots__ = ('_dynamic_group_identifier', '_states')
+
+    def __init__(self, host, modifier=None, modifier_blk=None,
+                 dynamic_group_identifier=None):
+        """Initialize radiance properties."""
+        _RadianceProperties.__init__(self, host, modifier, modifier_blk)
+        self._states = []
+        self.dynamic_group_identifier = dynamic_group_identifier
+
+    @property
+    def dynamic_group_identifier(self):
+        """Get or set a text string for the dynamic_group_identifier.
+        
+        Objects sharing the same dynamic_group_identifier will have their
+        states change in unison. If None, the object is assumed to be static.
+        """
+        return self._dynamic_group_identifier
+
+    @dynamic_group_identifier.setter
+    def dynamic_group_identifier(self, identifier):
+        if identifier is not None:
+            identifier = valid_rad_string(identifier)
+        else:
+            assert len(self._states) == 0, 'dynamic_group_identifier cannot be ' \
+                'None while states are assigned. Use the remove_states method ' \
+                'before setting to None.'
+        self._dynamic_group_identifier = identifier
+
+    @property
+    def states(self):
+        """Get or set an array of radiance states assigned to this object.
+    
+        These cannot be set unless there is also a dynamic_group_identifier for
+        the object.
+        """
+        return tuple(self._states)
+
+    @states.setter
+    def states(self, value):
+        if value is not None:
+            assert self._dynamic_group_identifier is not None, 'Object must have ' \
```

Comment:
Now that I see how this requirement affects the code maybe the idea of not enforcing an identifier would be a good idea.

We can set-up the identifier when they are grouped or instead of using a list of states here create an object for `States` which takes the states and the identifier as inputs. I see that you have a similar class in `dynamic` module. Still need some time to get my head around the overall hierarchy/workflow.

Reply:
No, I think you were right to suggest that the user should specify the dynamic group in order to be able to assign states. I think it's more explicit to ask the API user for this identifier than auto-generate whenever the states are set.